saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/util/SAMLParserUtil.java

```diff
@@ -316,25 +334,49 @@ public static Object parseAttributeValue(XMLEventReader xmlEventReader) throws P
                 return "";
             }
 
-            throw logger.unsupportedType(StaxParserUtil.getStartElementName(startElement));
+            // when no type attribute assigned -> assume anyType
+            return parseAnyTypeAsString(xmlEventReader);
         }
         //      RK Added an additional type check for base64Binary type as calheers is passing this type
         String typeValue = StaxParserUtil.getAttributeValue(type);
         if (typeValue.contains(":string")) {
             return StaxParserUtil.getElementText(xmlEventReader);
         } else if (typeValue.contains(":anyType")) {
-            // TODO: for now assume that it is a text value that can be parsed and set as the attribute value
-            return StaxParserUtil.getElementText(xmlEventReader);
+            return parseAnyTypeAsString(xmlEventReader);
         } else if(typeValue.contains(":base64Binary")){
             return StaxParserUtil.getElementText(xmlEventReader);
         } else if(typeValue.contains(":boolean")){
             return StaxParserUtil.getElementText(xmlEventReader);
         }
 
-
         throw logger.parserUnknownXSI(typeValue);
     }
 
+    public static String parseAnyTypeAsString(XMLEventReader xmlEventReader) throws ParsingException {
+        try {
+            XMLEvent event = xmlEventReader.peek();
+            if (event.isStartElement()) {
+                event = xmlEventReader.nextTag();
+                StringWriter sw = new StringWriter();
+                XMLEventWriter writer = XMLOutputFactory.newInstance().createXMLEventWriter(sw);
+                QName tagName = event.asStartElement().getName();
+                boolean endTagReached = false;
+                do {
+                    writer.add(event);
+                    event = (XMLEvent) xmlEventReader.next();
+                    endTagReached = event.isEndElement() && event.asEndElement().getName().equals(tagName);
+                } while (xmlEventReader.hasNext() && !endTagReached);
```

Comment:
Would the following xml snippet be processed correctly?
```xml
<status>
  <code>
    <status>XYZ</status>
  <code>
</status>
```
Please add a testcase for that.